packages/libs/src/sdk/services/AntiAbuseOracleSelector/AntiAbuseOracleSelector.ts

```diff
@@ -0,0 +1,57 @@
+import { sampleSize } from 'lodash'
+import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
+import type { LoggerService } from '../Logger'
+import { defaultAntiAbuseOracleSelectorConfig } from './constants'
+import type {
+  AntiAbuseOracleSelectorService,
+  AntiAbuseOracle,
+  AntiAbuseOracleSelectorConfig
+} from './types'
+
+export class AntiAbuseOracleSelector implements AntiAbuseOracleSelectorService {
+  private services: AntiAbuseOracle[] | null = null
+  private readonly logger: LoggerService
+  private readonly endpoints: string[]
+  private readonly addresses: string[]
+  constructor(config?: AntiAbuseOracleSelectorConfig) {
+    const configWithDefaults = mergeConfigWithDefaults(
+      config,
+      defaultAntiAbuseOracleSelectorConfig
+    )
+    this.endpoints = configWithDefaults.endpoints
+    this.addresses = configWithDefaults.addresses
+    this.logger = configWithDefaults.logger
+  }
+
+  public async getSelectedService() {
+    const services = await this.getHealthyRegisteredServices()
+    return sampleSize(services, 1)[0] ?? null
+  }
+
+  private async getHealthyRegisteredServices() {
+    if (this.services === null) {
+      this.services = []
+      for (const endpoint of this.endpoints) {
+        try {
+          const response = await fetch(`${endpoint}/health_check`)
```

Comment:
issue with `Promise.all` is that if one fails, then the whole call fails. and in this case we can't rely on other calls passing. meaning, beyond `response.ok` being falsy, there may not even be a response.

Reply:
Oops, thought this _was_ using `Promise.all`! Good call. I'll refactor this - to remedy your concern @sddioulde I think I can still try/catch and skip

Reply:
Re: the rejection, you can use `allSettled` instead, which will wait for all of them and then return an array of results without the early bail on first rejection

Reply:
yep. may still have to use a shim for that, unless all browsers now have support for it. this wasnt the case last year or so

Reply:
I think what we're after is actually `Promise.any()` which we already have a shim for and use in DN selector, which I should have used a similar model instead of being lazy. Updated to use that sort of style instead.